app/response_processor.py

```diff
@@ -0,0 +1,96 @@
+from app import receipt
+from app import settings
+from app.settings import session
+import logging
+from structlog import wrap_logger
+from requests.packages.urllib3.exceptions import MaxRetryError
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def remote_call(request_url, json=None, data=None):
+    try:
+        logger.info("Calling service", request_url=request_url)
+        r = None
+
+        if json:
+            r = session.post(request_url, json=json)
+        elif data:
+            r = session.post(request_url, data=data)
+        else:
+            r = session.get(request_url)
+
+        return r
+
+    except MaxRetryError:
+        logger.error("Max retries exceeded (5)", request_url=request_url)
+
+
+def response_ok(res):
+    if res.status_code == 200:
+        logger.info("Returned from service", request_url=res.url, status_code=res.status_code)
+        return True
+
+    else:
+        logger.error("Returned from service", request_url=res.url, status_code=res.status_code)
+        return False
+
+
+class ResponseProcessor:
+    def __init__(self, logger, skip_receipt=True):
+        self.logger = logger
+        self.skip_receipt = skip_receipt
+
+    def process(self, encrypted_survey):
+        # decrypt
+        decrypt_ok, decrypted_json = self.decrypt_survey(encrypted_survey)
+        if not decrypt_ok:
+            self.logger.error("Unable to decrypt survey")
+            return False
+
+        metadata = decrypted_json['metadata']
+        bound_logger = self.logger.bind(user_id=metadata['user_id'], ru_ref=metadata['ru_ref'])
+
+        # validate
+        validate_ok = self.validate_survey(decrypted_json)
```

Comment:
Where we're calling the response_ok method (which validate_survey does indirectly), we don't need to log there was an error in the bound_logger as response_ok should have done so already. Similarly for validate/store/receipt calls.